topics/tutorials/file_and_code_templates/using_file_templates.md

```diff
@@ -0,0 +1,131 @@
+service providing methods returning all or single file templates from a given category.
+For example, to obtain a template from the _[Code](providing_file_templates.md#code)_ category, pass its name to the `getCodeTemplate()` method (notice the lack of the `.ft` extension):
+
+```java
+FileTemplate template = FileTemplateManager.getInstance(project)
+    .getCodeTemplate("Test Class.java");
+```
+
+To render a template content, prepare and pass `Properties` object to the `getText()` method:
+
+```java
+Properties properties = new Properties();
+properties.setProperty("PROP1", value1);
+properties.setProperty("PROP2", value2);
+String renderedText = template.getText(properties);
+```
+
+## Creating New Files from Template
+
+The common use case for file templates is creating new files with the initial content specific to a language or framework supported by the plugin.
+File templates assigned to the _[Files](providing_file_templates.md#files)_ category are automatically available in the <menupath>File | New</menupath> action group.
+Sometimes, creating a file from a given template in a specific project place doesn't make sense, or a template requires some additional properties for its content.
+It is possible to control a file template's visibility and its available properties using
+[`CreateFromTemplateHandler`](upsource:///platform/lang-impl/src/com/intellij/ide/fileTemplates/CreateFromTemplateHandler.java)
+implementation registered in the `com.intellij.createFromTemplateHandler` EP.
+
+**Example**:
+[`JavaCreateFromTemplateHandler`](upsource:///java/java-impl/src/com/intellij/ide/fileTemplates/JavaCreateFromTemplateHandler.java)
+
+### Exposing File Templates from the Other Category
+
+File templates from the _[Other](providing_file_templates.md#other)_ category are not exposed by default.
+To make them available in the UI, a plugin has to implement and register an action.
+The easiest way to do it is by extending
+[`CreateFileFromTemplateAction`](upsource:///platform/lang-impl/src/com/intellij/ide/actions/CreateFileFromTemplateAction.java)
+base class which contains standard implementations of necessary methods.
+It allows customizing the action by providing an action name, icon, overriding methods creating dialog, and others.
+Example action:
+
+```java
+public class CreateMyClassAction extends CreateFileFromTemplateAction {
+
+  public CreateMyClassAction() {
+    super("My Class", "Creates new class", CLASS_ICON);
+  }
+
+  @Override
+  protected void buildDialog(Project project, PsiDirectory directory,
+      CreateFileFromTemplateDialog.Builder builder) {
+    builder
+    .setTitle("New My File")
+    .addKind("Class", CLASS_ICON, "My Class");
+  }
+
+  @Override
+  protected String getActionName(PsiDirectory directory,
+      @NotNull String newName, String templateName) {
+    return "Create My Class: " + newName;
+  }
+}
+```
+
+The new action should be registered under the `NewGroup` group, e.g:
```

Comment:
`<menupath>`

Reply:
This one is group ID, not menu path. I think it's correct.

Reply:
I agree with Karol. The style guide says

> Use the menupath tag to specify path to a menu option.

which it is not. We can fix that later if Yann persists, but it's the only blocking change request.

Reply:
right, sorry